presto-main/src/test/java/io/prestosql/type/TestUUIDOperators.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.type;
+
+import io.airlift.slice.Slices;
+import io.prestosql.operator.scalar.AbstractTestFunctions;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.type.SqlVarbinary;
+import org.testng.annotations.Test;
+
+import static com.google.common.io.BaseEncoding.base16;
+import static io.prestosql.spi.function.OperatorType.HASH_CODE;
+import static io.prestosql.spi.function.OperatorType.INDETERMINATE;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.VarbinaryType.VARBINARY;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static io.prestosql.type.UUIDType.UUID;
+
+public class TestUUIDOperators
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testVarcharToUUIDCast()
+    {
+        assertFunction("CAST('1-2-3-4-1' AS UUID)", UUID, "00000001-0002-0003-0004-000000000001");
```

Comment:
These shorter string representations don’t seem to be allowed by RFC 4122. I think the fact that Java accepts it is a bug caused by an implementation detail.

Let’s enforce the input length. Add tests for multiple UUID versions, plus the nil UUID.

Reply:
For the documentation, let's make sure we mention the RFC.

Reply:
Have added this to types documentation and also handled for various version of UUID